### DISCUSSION TITLE
Deepspeed ZeRo3 Error 'RuntimeError: 'weight must be 2-D' Please help

### DISCUSSION THREAD

**Discussion by shing100** [deepspeedai/DeepSpeed#6638] [Q&A]
Hello, deep speed users

I'm trying to learn from github code (https://github.com/arcee-ai/DistillKit/blob/main/distil_hidden.py) using Zero3 settings, but I get an error called "RuntimeError: 'weight must be 2-D'.

Can you help me?

config
```
{
  "zero_force_ds_cpu_optimizer": false,
  "zero_allow_untested_optimizer": true,
  "zero_optimization": {
    "stage": 3,
    "offload_param": {
      "device": "cpu",
      "pin_memory": true
    },
    "overlap_comm": true,
    "contiguous_gradients": true,
    "sub_group_size": 0,
    "reduce_bucket_size": "auto",
    "stage3_prefetch_bucket_size": "auto",
    "stage3_param_persistence_threshold": "auto",
    "stage3_max_live_parameters": 0,
    "stage3_max_reuse_distance": 0,
    "stage3_gather_16bit_weights_on_model_save": true
  },
  "bf16": {
    "enabled": true
  },
  "fp16": {
    "enabled": "auto",
    "auto_cast": false,
    "loss_scale": 0,
    "initial_scale_power": 32,
    "loss_scale_window": 1000,
    "hysteresis": 2,
    "min_loss_scale": 1
  },
  "gradient_accumulation_steps": "auto",
  "gradient_clipping": "auto",
  "train_batch_size": "auto",
  "train_micro_batch_size_per_gpu": "auto",
  "wall_clock_breakdown": false
}
```

**Comment by dhingratul** [comment]
Any resolution on this ?

**Comment by tjruwase** [comment]
@dhingratul, can you share full stack trace? I suspect the problem is that partitioned parameters are not gathered before access: 

1. https://deepspeed.readthedocs.io/en/latest/zero3.html#manual-parameter-coordination
2. https://deepspeed.readthedocs.io/en/latest/zero3.html#manual-parameter-coordination

Such problems are often solved using the parameter gathering context manager like
https://github.com/deepspeedai/DeepSpeedExamples/blob/223665c37a6fa8c0c3e916aebb4c52d336b046b5/applications/DeepSpeed-Chat/dschat/utils/utils.py#L276
